### PR TITLE
Remove support for DuplicatePolicy

### PR DESCRIPTION
**Pull request by codingllama**
Handling duplicates is a personality responsibility - if the personality wants to allow duplicates, then it should create non-conflicting identity hashes.

### CODE REVIEW

**Comment by codingllama** [comment]
I've kept functionality as-is, so the PR is as focused as possible. I plan to address some other related TODOs (such as validating tree types, getting hash configs, etc) in follow up PRs.

File: README.md [hunk 1]
```diff
@@ -135,6 +135,7 @@ Some of the Trillian Go code is autogenerated from other files:
    definitions.
  - Some unit tests use mock implementations of interfaces; these are created
    from the real implementations by [GoMock](https://github.com/golang/mock).
+ - stringer (`go get golang.org/x/tools/cmd/stringer`)
```
Comment (daviddrysdale) [inline README.md]: A sentence rather than just a name would be nice.

File: trillian.pb.go [hunk 2]
```diff
@@ -114,35 +114,6 @@ func (x TreeType) String() string {
 }
 func (TreeType) EnumDescriptor() ([]byte, []int) { return fileDescriptor3, []int{2} }
 
-// Duplicate policy of a tree.
```
Comment (daviddrysdale) [inline trillian.pb.go]: This change looks like it's in the wrong commit.

File: scripts/presubmit.sh [hunk 3]
```diff
@@ -90,13 +91,14 @@ main() {
...
     echo "Missing license header in: ${nolicense}"
-    exit 5
+    exit 2
   fi
 
   if [[ "${run_generate}" -eq 1 ]]; then
     echo 'running go generate'
     go generate -run="protoc" ./...
     go generate -run="mockgen" ./...
+    go generate -run="stringer" ./...
```
Comment (daviddrysdale) [inline scripts/presubmit.sh]: Travis is complaining about something with this.

**Comment by Martin2112** [comment]
Originally handling duplicates wasn't a personality responsibility and we didn't have the leaf identity / merkle hash distinction. We changed our minds on this as we went along so this is probably the right thing to do now.

Can you open an issue for this if there isn't one?

**Comment by codingllama** [comment]
Sure, I'll raise an issue.

I had cleared this up previously with @AlCutter, but I felt he was too swamped to take the review. I should have mentioned this somewhere.

File: scripts/presubmit.sh (hunk 3 above)
Comment (codingllama) [inline scripts/presubmit.sh]: Done. I missed a line on .travis.yaml.

File: trillian.pb.go (hunk 2 above)
Comment (codingllama) [inline trillian.pb.go]: Ack.

File: README.md (hunk 1 above)
Comment (codingllama) [inline README.md]: Done.

**Comment by codingllama** [comment]
Fixes #510.

PTAL?

**Comment by daviddrysdale** [comment]
Could you rebase this into two distinct commits (fiddle with stringer, remove duplicates) and rebase-and-merge when you get that far please?

File: README.md [hunk 4]
```diff
@@ -135,7 +135,8 @@ Some of the Trillian Go code is autogenerated from other files:
    definitions.
  - Some unit tests use mock implementations of interfaces; these are created
    from the real implementations by [GoMock](https://github.com/golang/mock).
- - stringer (`go get golang.org/x/tools/cmd/stringer`)
+ - stringer fmt.Stringer support for enums
```
Comment (codingllama) [inline README.md]: Done.

**Comment by codingllama** [comment]
PTAL? Rebased as requested.

**Pull request merged**